app/controllers/api/messaging.php

```diff
@@ -0,0 +1,2961 @@
+                && \array_key_exists('endpoint', $credentials)
+            ) {
+                $enabled = true;
+            } else {
+                $enabled = false;
+            }
+
+            $provider->setAttribute('enabled', $enabled);
+        }
+
+        $provider = $dbForProject->updateDocument('providers', $provider->getId(), $provider);
+
+        $queueForEvents
+            ->setParam('providerId', $provider->getId());
+
+        $response
+            ->dynamic($provider, Response::MODEL_PROVIDER);
+    });
+
+App::delete('/v1/messaging/providers/:providerId')
+    ->desc('Delete provider')
+    ->groups(['api', 'messaging'])
+    ->label('audits.event', 'provider.delete')
+    ->label('audits.resource', 'provider/{request.$providerId}')
+    ->label('event', 'providers.[providerId].delete')
+    ->label('scope', 'providers.write')
+    ->label('sdk.auth', [APP_AUTH_TYPE_ADMIN, APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'messaging')
+    ->label('sdk.method', 'deleteProvider')
+    ->label('sdk.description', '/docs/references/messaging/delete-provider.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_NOCONTENT)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_NONE)
+    ->param('providerId', '', new UID(), 'Provider ID.')
+    ->inject('queueForEvents')
+    ->inject('dbForProject')
+    ->inject('response')
+    ->action(function (string $providerId, Event $queueForEvents, Database $dbForProject, Response $response) {
+        $provider = $dbForProject->getDocument('providers', $providerId);
+
+        if ($provider->isEmpty()) {
+            throw new Exception(Exception::PROVIDER_NOT_FOUND);
+        }
+
+        $dbForProject->deleteDocument('providers', $provider->getId());
+
+        $queueForEvents
+            ->setParam('providerId', $provider->getId());
+
+        $response
+            ->setStatusCode(Response::STATUS_CODE_NOCONTENT)
+            ->noContent();
+    });
+
+App::post('/v1/messaging/topics')
+    ->desc('Create a topic.')
+    ->groups(['api', 'messaging'])
+    ->label('audits.event', 'topic.create')
+    ->label('audits.resource', 'topic/{response.$id}')
+    ->label('event', 'topics.[topicId].create')
+    ->label('scope', 'topics.write')
+    ->label('sdk.auth', [APP_AUTH_TYPE_ADMIN, APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'messaging')
+    ->label('sdk.method', 'createTopic')
+    ->label('sdk.description', '/docs/references/messaging/create-topic.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_CREATED)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_TOPIC)
+    ->param('topicId', '', new CustomId(), 'Topic ID. Choose a custom Topic ID or a new Topic ID.')
+    ->param('name', '', new Text(128), 'Topic Name.')
+    ->param('description', '', new Text(2048), 'Topic Description.', true)
```

Comment:
Does the validator need to be updated to allow empty string?

Reply:
The test seems to work fine without description. If you are skipping the description property, the topic should be able to be created.